mempool/reactor.go

```diff
@@ -0,0 +1,317 @@
+package mempool
+
+import (
+	"errors"
+	"fmt"
+	"math"
+	"time"
+
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/libs/clist"
+	"github.com/tendermint/tendermint/libs/log"
+	tmsync "github.com/tendermint/tendermint/libs/sync"
+	"github.com/tendermint/tendermint/p2p"
+	protomem "github.com/tendermint/tendermint/proto/tendermint/mempool"
+	"github.com/tendermint/tendermint/types"
+)
+
+const (
+	MempoolChannel = byte(0x30)
+
+	peerCatchupSleepIntervalMS = 100 // If peer is behind, sleep this amount
+
+	// UnknownPeerID is the peer ID to use when running CheckTx when there is
+	// no peer (e.g. RPC)
+	UnknownPeerID uint16 = 0
+
+	maxActiveIDs = math.MaxUint16
+)
+
+// Reactor handles mempool tx broadcasting amongst peers.
+// It maintains a map from peer ID to counter, to prevent gossiping txs to the
+// peers you received it from.
+type Reactor struct {
+	p2p.BaseReactor
+	config  *cfg.MempoolConfig
+	mempool *CListMempool
+	ids     *mempoolIDs
+}
```

Comment:
wait a sec, we don't want the reactor in here, right?

Reply:
`git add mempool` was not a good way of doing this :D good catch!